### PR TITLE
Allow analyzing a boundary result from geocoder

### PR DESCRIPTION
**Pull request by arottersman**
## Overview

Implements behavior in [these wireframes](https://marvelapp.com/33efhfb/screen/31228062)

- when you search for and select a location, the selected location's name appears in the geocoder, and you are directed to `/draw`
- if the the selected location is a boundary layer, a `Select` button appears in the geocoder
- clicking the `Select` button analyzes the shape

#### Enhancement
- Pressing `ESC` now clears the selected shape and resets the query

Connects #2138 

### Demo
_Selecting a boundary_
xbgyajudai

_If you select a regular search result, displays the address_
screen shot 2017-08-14 at 3 27 07 pm

## Testing Instructions

 * Pull and `./scripts/bundle.sh`
 
**Address search**
* search for an address
* select a result
* confirm it appears in the search bar, that there are no errors in the console, and you are not directed to `/draw`
* press `ESC` and confirm the result clears
* search for something else and press `ENTER`. The first result should get selected

**Boundary search**
* search for and select a HUC
* Confirm you are taken to `/draw`
* Click `Select`
* Confirm you are taken to `/analyze` and the shape name, layer name, and wkaoi properties have come through (the names should be at the top of the sidebar)
* search again and this time press `ESC` --> the selected shape should clear from the map

### CODE REVIEW

File: src/mmw/js/src/draw/controllers.js [hunk 1]
```diff
@@ -18,9 +18,12 @@ var DrawController = {
             toolbarModel = new models.ToolbarModel();
 
         toolbarModel.set('predefinedShapeTypes',
-            _.filter(settings.get('boundary_layers'), { selectable: true }));
+                         _.filter(settings.get('boundary_layers'), { selectable: true }));
```
Comment (arottersman) [inline src/mmw/js/src/draw/controllers.js]: This auto-indented. I'll fix it.

**Comment by ajrobbins** [comment]
In the wireframes, the search result AoI has a red outline and is not reverse masked, which seems helpful if the user wants to inspect the area around the AoI and maybe change AoIs before selecting.

**Comment by arottersman** [comment]
> In the wireframes, the search result AoI has a red outline and is not reverse masked, which seems helpful if the user wants to inspect the area around the AoI and maybe change AoIs before selecting.

Thanks for catching that. I'll add a commit.

**Comment by arottersman** [comment]
This should be ready for another look.


wiwm21uyrw

**Comment by arottersman** [comment]
Thanks!

**Pull request merged**